velox/exec/TableWriter.cpp

```diff
@@ -99,11 +97,56 @@ RowVectorPtr TableWriter::getOutput() {
   }
   finished_ = true;
 
-  connector::CommitInfo commitInfo(
-      dataSink_->getConnectorCommitInfo(),
-      numWrittenRows_,
-      outputType_,
-      driverCtx_->task->taskId());
-  return writeProtocol_->commit(commitInfo, pool());
+  memory::MemoryPool* pool = connectorQueryCtx_->memoryPool();
+
+  if (outputType_->size() <= 1) {
+    return std::make_shared<RowVector>(
+        pool,
+        outputType_,
+        nullptr,
+        1,
+        std::vector<VectorPtr>{
+            BaseVector::createConstant((int64_t)numWrittenRows_, 1, pool)});
+  }
+
+  std::vector<std::string> fragments = dataSink_->finish();
+
+  std::vector<VectorPtr> columns = {};
```

Comment:
nit: consider moving this variable the end of the method and initialize it like so:

```
std::vector<VectorPtr> columns = {writtenRowsVector, fragmentsVector, commitContextVector};
```